src/WebJobs.Script/Binding/Http/HttpExtensions.cs

```diff
@@ -0,0 +1,284 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Script.Binding
+{
+    public static class HttpExtensions
```

Comment:
Let's not add these proxy specific helpers as extension methods. We don't want them to show up in general for all http requests. I think they'd be better if they were just static helper methods on a utility class for your proxy stuff.